config/src/iroha.rs

```diff
@@ -0,0 +1,153 @@
+//! This module contains [`Configuration`] structure and related implementation.
+use std::{fmt::Debug, fs::File, io::BufReader, path::Path};
+
+use eyre::{Result, WrapErr};
+use iroha_config_base::derive::{view, Configurable};
+use iroha_crypto::prelude::*;
+use serde::{Deserialize, Serialize};
+
+use super::*;
+
+// Generate `ConfigurationView` without the private key
+view! {
+    /// Configuration parameters for a peer
+    #[derive(Debug, Clone, Deserialize, Serialize, Configurable)]
+    #[serde(default)]
+    #[serde(rename_all = "UPPERCASE")]
+    #[config(env_prefix = "IROHA_")]
+    pub struct Configuration {
+        /// Public key of this peer
+        #[config(serde_as_str)]
+        pub public_key: PublicKey,
+        /// Private key of this peer
+        #[view(ignore)]
+        pub private_key: PrivateKey,
+        /// Disable coloring of the backtrace and error report on panic
+        pub disable_panic_terminal_colors: bool,
+        /// Iroha will shutdown on any panic if this option is set to `true`.
+        pub shutdown_on_panic: bool,
+        /// `Kura` configuration
+        #[config(inner)]
+        pub kura: kura::Configuration,
+        /// `Sumeragi` configuration
+        #[config(inner)]
+        #[view(into = sumeragi::ConfigurationView)]
+        pub sumeragi: sumeragi::Configuration,
+        /// `Torii` configuration
+        #[config(inner)]
+        pub torii: torii::Configuration,
```

Comment:
Hm, I think it's quite dangerous. We have to remember which configuration contains private keys and replace them with their `View` analogue.
I have an idea how to bypass this, but it may be too complex and probably we can do it in another issue.

The idea is the next. When using `view!` let's implement a new marker trait, i.e. called `PrivateData`, for the original struct. Then in another `view!` call we can check if a type of a field implements such trait (there are ways to do that). Unfortunately at compile-time we know nothing about time except his name, so this check will be at runtime. I.e. in `From` implementation. If some filed's type implements `PrivateData` then panic.

What to you think?

Reply:
I'm totally agree that converting to `View` automatically would be great feature to have.
However i'm not sure that proposed solution would work, problem is that we must provide `struct` field types at compile time, or i miss something?
Problem would be easily solved if only rust has [negative_impls](https://doc.rust-lang.org/beta/unstable-book/language-features/negative-impls.html).

Reply:
Yes, we have to provide them in compile time and we can automatically find the error only in runtime. But it's better not to find error at all, I think.

Reply:
oh, i get what you are suggesting, we could use `static_assertion` for this purpose